keycloak/client.go

```diff
@@ -286,6 +294,21 @@ func (c Client) ListGroups(ctx context.Context) ([]Group, error) {
 		return nil, err
 	}
 
+	for _, g := range groups {
+		logger := log.FromContext(ctx)
+		subgroups, err := c.getChildGroups(ctx, token, *g.ID)
```

Comment:
On a second thought: Isn't there a risk we delete something if we get intermittent 404s for whatever reason and subgroups is empty?

Reply:
Maybe we could use [`GoCloak.GetServerInfo`](https://pkg.go.dev/github.com/Nerzal/gocloak/v13#GoCloak.GetServerInfo) ?

Reply:
I think a "disable KC 22 fallback" toggle would be enough.